analysis/bulk-deconvolution/exploratory-notebooks/explore-quantiseq-results.Rmd

```diff
@@ -0,0 +1,348 @@
+
+We'll make histogram showing the proportion of classified cells for samples in each project:
+
+```{r}
+classified_df <- quantiseq_df |>
+  dplyr::filter(is_other) |>
+  dplyr::mutate(classified_proportion = 1 - proportion) 
+
+sample_order <- classified_df |>
+  dplyr::group_by(project_diagnosis) |>
+  dplyr::arrange(classified_proportion) |>
+  dplyr::pull(sample_id)
+```
+
+
+```{r fig.height=4, fig.width=10}
+make_barplot(
+  classified_df,
+  xvar = forcats::fct_relevel(sample_id, sample_order), 
+  yvar = classified_proportion, 
+  facetvar = project_diagnosis
+) + 
+  xlab("Sample arranged by classified proportion") 
+```
+
+
+The percentage of classified cells in each sample ranges from close to 0 to up to around 0.55.
+`SCPCP000001` and `SCPCP000002` tend to have the highest values.
+
+Which samples have lower classified proportions?
+
+```{r}
+classified_df |>
+  dplyr::filter(classified_proportion <= 0.05) |>
+  dplyr::select(project_id, sample_id, classified_proportion, diagnosis) |>
+  dplyr::arrange(classified_proportion)
+```
+
+### Proportions of cell types
+
+If we consider _only_ cells which were classified, what proportions of cell types are represented across samples?
+
+First, we'll need to remove the "Other" cells and normalize the remaining values to sum to 1.
+
+```{r}
+classified_normalized_df <- quantiseq_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::group_by(sample_id) |>
+  # renormalize so non-other sums to 1
+  dplyr::mutate(proportion = proportion / sum(proportion))
+```
+
+
+```{r fig.height=5, fig.width=12}
+make_barplot(
+  classified_normalized_df,
+  xvar = sample_id, 
+  yvar = proportion, 
+  fillvar = quantiseq_celltype, 
+  facetvar = project_diagnosis
+)
+```
+
+Very generally speaking, we see:
+
+- `SCPCP000001` and `SCPCP000002` have relatively higher proportions of `Dendritic cells` compared to other projects
+- `SCPCP000006` and `SCPCP000009` have relatively higher proportions of `B.cells` compared to other projects
+- `SCPCP000017` appears to most frequently contain `Macrophages.M2` compared to other projects
+- `T.cells.CD4` and `T.cells.CD8` appear fairly infrequently,  although the `quanTIseq` paper does point out 
+
+Let's look at some associated summaries for this sample-specific information.
+
+
+Of the 10 cell types in `quanTIseq`, how many are represented in each sample?
+For this, we'll make a histogram of the number of cell types identified per sample (regardless of their proportion).
+
+```{r}
+celltype_counts_df <- quantiseq_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::group_by(project_diagnosis, sample_id) |>
+  dplyr::summarize(num_celltypes = sum(proportion != 0))
+```
+
+```{r fig.height=4, fig.width=10}
+ggplot(celltype_counts_df) + 
+  aes(x = num_celltypes) + 
+  geom_histogram(center = 0, binwidth = 1) + 
```

Comment:
This should make integer ticks, since integers.
```suggestion
  geom_histogram(center = 0, binwidth = 1) + 
  scale_x_continuous(breaks = scales::breaks_pretty()) +
```